uplink/config.go

```diff
@@ -39,8 +39,8 @@ type ClientConfig struct {
 	SatelliteAddr  string        `releaseDefault:"127.0.0.1:7777" devDefault:"127.0.0.1:10000" help:"the address to use for the satellite" noprefix:"true"`
 	MaxInlineSize  memory.Size   `help:"max inline segment size in bytes" default:"4KiB"`
 	SegmentSize    memory.Size   `help:"the size of a segment in bytes" default:"64MiB"`
-	RequestTimeout time.Duration `help:"timeout for request" default:"0h0m20s"`
-	DialTimeout    time.Duration `help:"timeout for dials" default:"0h0m20s"`
+	RequestTimeout time.Duration `help:"timeout for request" default:"0h2m00s"`
+	DialTimeout    time.Duration `help:"timeout for dials" default:"0h2m00s"`
```

Comment:
These don't match with `DefaultRequestTimeout` and `DefaultDialTimeout`. Without digging through stuff, is there an easy answer to where this would apply versus where the DefaultFoo versions apply?

Reply:
It's not very obvious but these defaults get applied via `UplinkFlags` here in [cmd/uplink/cmd/root.go](https://github.com/storj/storj/blob/master/cmd/uplink/cmd/root.go#L30). Since the Satellite goes down a different path it gets a different default applied from `pkg/transport/common.go` above.

Reply:
So these defaults apply to things using the Config struct, while the ones in pkg/transport/common.go only apply for the satellite? Can we move the satellite-only defaults in to the satellite-specific code? Or if it happens by some other mechanism that other code might need to use eventually, can we at least clarify in the comments for `DefaultRequestTimeout` and `DefaultDialTimeout` where they apply as opposed to the default Config values?

Reply:
I added the comments you suggested. The config struct only applies to the uplink CLI. `pkg/transport/common.go` applies to every other use case in various packages such as `satellite`, `bootstrap`, `testplanet`, `certificates`, `kademlia`, `peertls`, `storagenode`. I'm not sure pulling it out into the Satellite makes sense yet because all of these still need to inherit a default timeout. Any consumers of libuplink need defaults as well.

Maybe we override it as needed in the future?

Reply:
I don't see the added comments, but a grep for `uplink.Config` shows it being used in `cmd/uplink`, `cmd/gateway`, `testplanet`, and `miniogw` at least. `testplanet` then might use _both_? It seems like making `DefaultDialTimeout` and `DefaultRequestTimeout` public has made the situation a lot more confusing. Or maybe I just haven't identified the right logical separation between these?

Reply:
Yeah, it did make it a little bit more confusing than it already was. I was trying to get a fix out. We use the transport in different ways. Sometimes from an uplink client but sometimes direct. For example. It was mentioned above to make `DefaultDialTimeout` public because it may be cleaner to share the transport default instead of just duplicating it somewhere else in the code.

Do you have any other suggestions?

The callstack patterns look as follows.

#### Gateway 
```
lib/uplink.NewUplink()
        lib/uplink.setDefaults()
                  applies `transport.DefaultDialTimeout of 20s
                          passes into `transport.NewClientWithTimeouts()
```

#### Minio
Same as Gateway.
```
lib/uplink.NewUplink()
        lib/uplink.setDefaults()
                  applies `transport.DefaultDialTimeout of 20s
                          passes into `transport.NewClientWithTimeouts()
```

#### Satellite
```
transport.NewClient()
        transport.NewClientWithTimeouts() passed no timeouts
                defaults from transport.DefaultRequestTimeout gets applied.
```

#### Storage node
Same as above Satellite
```
transport.NewClient()
        transport.NewClientWithTimeouts()`passed no timeouts.
                defaults from transport.DefaultRequestTimeout gets applied.
```

#### Test Planet
Explicitly sets up a config and sets the following.
```
config.Client.RequestTimeout = 10 * time.Second
config.Client.DialTimeout = 10 * time.Second

lib/uplink.NewUplink() passed with a timeout of 10s
       passes into `transport.NewClientWithTimeouts()
```

Reply:
When defaultRequestTimeout was private, it was easy to tell that it only applied to that package, and easy to find where specifically it was used. Now it seems like I have to dig through code to find call stacks in order to determine whether we get the default default or the other default.

What if we had `defaultTransportRequestTimeout` in pkg/transport, a separate `defaultUplinkRequestTimeout` in lib/uplink, and the existing config-item default in uplink/config.go, and just let them all be different things?

Reply:
I'm interested in the resolution and/or the answer to @thepaul's last question but I'm not opposed to that going into another ticket/PR.